src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -0,0 +1,444 @@
+                    currentHash = hash
+                        .Aggregate(new StringBuilder(), (a, b) => a.Append(b.ToString("x2")))
+                        .ToString();
+                }
+
+                // get the last hash value if present
+                string lastHash = null;
+                if (await hashBlob.ExistsAsync())
+                {
+                    lastHash = await hashBlob.DownloadTextAsync();
+                    _logger.LogDebug($"SyncTriggers hash (Last='{lastHash}', Current='{currentHash}')");
+                }
+
+                if (string.Compare(currentHash, lastHash) != 0)
+                {
+                    // hash will need to be updated - return the
+                    // new hash value
+                    return currentHash;
+                }
+            }
+            catch (Exception ex)
+            {
+                // best effort
+                _logger.LogError(0, ex, "Error checking SyncTriggers hash");
+            }
+
+            // if the last and current hash values are the same,
+            // or if any error occurs, return null
+            return null;
+        }
+
+        internal async Task UpdateHashAsync(CloudBlockBlob hashBlob, string hash)
+        {
+            try
+            {
+                // hash value has changed or was not yet stored
+                // update the last hash value in storage
+                await hashBlob.UploadTextAsync(hash);
+                _logger.LogDebug($"SyncTriggers hash updated to '{hash}'");
+            }
+            catch (Exception ex)
+            {
+                // best effort
+                _logger.LogError(0, ex, "Error updating SyncTriggers hash");
+            }
+        }
+
+        internal CloudBlockBlob GetHashBlob()
+        {
+            if (_hashBlob == null)
+            {
+                string hostId = _hostConfig.HostConfig.HostId;
+                string storageConnectionString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
+                CloudStorageAccount account = null;
+                if (!string.IsNullOrEmpty(storageConnectionString) &&
+                    CloudStorageAccount.TryParse(storageConnectionString, out account) && !string.IsNullOrEmpty(hostId))
+                {
+                    CloudBlobClient blobClient = account.CreateCloudBlobClient();
+                    var blobContainer = blobClient.GetContainerReference(ScriptConstants.AzureWebJobsHostsContainerName);
+                    
+                    string hashBlobPath = $"synctriggers/{hostId}/last";
+                    _hashBlob = blobContainer.GetBlockBlobReference(hashBlobPath);
+                }
+            }
+            return _hashBlob;
+        }
+
+        public async Task<JToken> GetSyncTriggersPayload()
+        {
+            var functionsMetadata = WebFunctionsManager.GetFunctionsMetadata(_hostConfig, logger: null);
```

Comment:
Will revert this "logger: null" change and pass an actual logger down. Initially I was thinking of suppressing all the file system logs for sync triggers, but I guess they don't hurt.